rotkehlchen/tests/utils/history_base_entry.py

```diff
@@ -148,35 +217,13 @@ def add_entries(server: 'APIServer', events_db: DBHistoryEvents, add_directly: b
         depositor=string_to_evm_address('0x0EbD2E2130b73107d0C45fF2E16c93E7e2e10e3a'),
     )]
 
-    entries_added_using_api = 0
     for entry in entries:
-        if (
-            add_directly is True or
-            isinstance(entry, (EthBlockEvent, EthDepositEvent, EthWithdrawalEvent, HistoryEvent)) or  # noqa: E501
-            (isinstance(entry, EvmEvent) and entry.extra_data is not None)
-        ):
-            # If any entry has extra data add it directly to the database instead of
-            # making use of the API. The reason is that the API doesn't allow to edit
-            # the extra data field.
-            # Also add HistoryEvent since the API doesn't allow to add them atm
-            with events_db.db.conn.write_ctx() as write_cursor:
-                identifier = events_db.add_history_event(
-                    write_cursor=write_cursor,
-                    event=entry,
-                )
-                entry.identifier = identifier
-        else:
-            json_data = entry_to_input_dict(entry, include_identifier=False)  # type: ignore
-            response = requests.put(
-                api_url_for(server, 'historyeventresource'),
-                json=json_data,
-            )
-            result = assert_proper_response_with_result(response)
-            assert 'identifier' in result
-            entry.identifier = result['identifier']
-            entries_added_using_api += 1
-
-    if add_directly is False:
-        assert entries_added_using_api == 4, 'api was used directly and event is missing'
-
+        json_data = entry_to_input_dict(entry, include_identifier=False)
+        response = requests.put(
+            api_url_for(server, 'historyeventresource'),
+            json=json_data,
+        )
+        result = assert_proper_response_with_result(response)
+        assert 'identifier' in result
```

Comment:
this is doing an http request, exceuting the validation logic... it is doing more things than what you need. If you want only to add the events call the method directly from the db object instead